internal/services/serviceprincipals/client_config_data_source_msgraph.go

```diff
@@ -0,0 +1,22 @@
+package serviceprincipals
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+)
+
+func clientConfigDataSourceReadMsGraph(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+	objectId := client.Claims.ObjectId
+	d.SetId(fmt.Sprintf("%s-%s-%s", client.TenantID, client.ClientID, objectId))
```

Comment:
Might be worth making the ID here just the Tenant ID, since the Client ID and Object ID would likely change through other invocations. That's allow users who are using different accounts (e.g. CI and local) against the same tenant to show changes and not force-new resources if the values/ID changes?

Reply:
Granted you can end up with diff flapping if you run Terraform with different auth principals, but this is also commonly used to assign object ownership in other resources and if that ends up mismatched due to running with a different principal then Terraform can lock itself out of other resources.